src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -300,30 +300,46 @@ private string InsertDeletedRange(DocumentsOperationContext context, DeletionRan
                 (changeVector, etag) = GenerateChangeVector(context);
             }
 
-            using (var sliceHolder = new TimeSeriesSliceHolder(context, documentId, name, collectionName.Name).WithEtag(etag))
-            using (table.Allocate(out var tvb))
-            using (Slice.From(context.Allocator, changeVector, out var cv))
+            var hash = (long)Hashing.XXHash64.Calculate(changeVector, Encoding.UTF8);
+
+            using (var sliceHolder = new TimeSeriesSliceHolder(context, documentId, name, collectionName.Name).WithChangeVectorHash(hash))
             {
-                tvb.Add(sliceHolder.TimeSeriesKeySlice);
-                tvb.Add(Bits.SwapBytes(etag));
-                tvb.Add(cv);
-                tvb.Add(sliceHolder.CollectionSlice);
-                tvb.Add(context.GetTransactionMarker());
-                tvb.Add(from.Ticks);
-                tvb.Add(to.Ticks);
+                if (table.SeekOneBackwardByPrimaryKeyPrefix(sliceHolder.TimeSeriesPrefixSlice, sliceHolder.TimeSeriesKeySlice, out var segmentValueReader) ||
+                    table.SeekOnePrimaryKeyWithPrefix(sliceHolder.TimeSeriesPrefixSlice, sliceHolder.TimeSeriesKeySlice, out segmentValueReader))
```

Comment:
why do we need to search the same table twice?

Reply:
done